src/Perpetuum/Services/EventServices/EPBonusEventService.cs

```diff
@@ -64,15 +116,35 @@ public override void Update(TimeSpan time)
             if (_endingEvent)
```

Comment:
`_eventStarted` and `_endingEvent` need protecting by a read lock, or by moving the write lock up to line 113.